### PR TITLE
Add a basic overview of the app catalog.

### PR DESCRIPTION
**Pull request by oponder**
@puja108 

Here's a start. This means we'd have two pages. One introducing the basic overview and concepts, and then another in the reference section showing some Happa screenshots and explanation of how to do the most common tasks.

Possibly a third page is needed telling more about the SLA of the managed apps and the introducing the incubator catalog and explaining how apps go from on to the other.

And perhaps another page to explain the details of app versioning and upgrading, though I think that could also live in the first overview document, depending on how complex that gets.

### CODE REVIEW

File: src/content/basics/app-catalog/index.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+In the near future you might also see _Apps_ provided by our partners.
+
+These collections of _Apps_ are grouped in _App Catalogs_, which in turn are
+browseable through our web interface.
+
+So to clarify a bit: the _Giant Swarm App Catalog_ refers to the whole feature,
+and an _App Catalog_ is a collection of _Apps_.
+
+We'll provide some _App Catalogs_, but so will our partners, and you too can set
+up your own internal catalog for your whole company to use.
```
Comment (puja108) [inline src/content/basics/app-catalog/index.md]:
```suggestion
up your own internal catalog(s) for your whole company to use.
```
as there can be more

File: src/content/basics/app-catalog/index.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+For example, the "App" Custom Resource indicates that you want an App installed
+on a specific tenant cluster.
+
+SNIPPET SHOWING A APP CR
+
+Below you can see a high level overview of the components and resources that work
+together to enable the features of the Giant Swarm App Catalog:
+
+SOME KIND OF USEFUL IMAGE SHOWING OPERATORS AND CRS AND SUCH.
```
Comment (puja108) [inline src/content/basics/app-catalog/index.md]: we can use a simplified version of what we have in miro and our internal spec right now

File: src/content/basics/app-catalog/index.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+### What are Managed Apps?
+
+Managed Apps are provided by Giant Swarm and have a SLA associated with
```
Comment (puja108) [inline src/content/basics/app-catalog/index.md]:
```suggestion
Managed Apps are provided by Giant Swarm and have an SLA associated with
```

grammarly fix

File: src/content/basics/app-catalog/index.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+them. When you install one of our Managed Apps, we make sure that it stays
+running and even get paged if something goes wrong.
```
Comment (puja108) [inline src/content/basics/app-catalog/index.md]: Maybe add that this is similar to how it already works for NGINX Ingress Controller and CoreDNS

File: src/content/basics/app-catalog/index.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+Managed Apps can be found in the Managed _App Catalog_, indicated by the
```
Comment (puja108) [inline src/content/basics/app-catalog/index.md]: this is happa right? maybe we want to mention this

File: src/content/basics/app-catalog/index.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+orange "Managed" badge.
+
+MORE DETAILS ON THE SLA AND EXPECTATION MANAGEMENT
+
+###
+
+### How can I interact with the App Catalog?
+
+Currently only through our web interface. See [this reference page] for more
```
Comment (puja108) [inline src/content/basics/app-catalog/index.md]: and API right?

File: src/content/basics/app-catalog/index.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+details on how to install and configure apps using our web interface.
+
+Support for the App Catalog is coming to our CLI (gsctl) next.
+
+Lastly, at the end of the day, the App Catalog is configured through a set of
+Custom Resources installed on your Control Plane.
+
+Our intention is to eventually grant you full access to the control plane
+API, and all of these resources, so that you can interact with them using
+`kubectl`, and automate them just as you have been automating other parts of your
```
Comment (puja108) [inline src/content/basics/app-catalog/index.md]:
```suggestion
`kubectl` (think `kubectl create app`), and automate them just as you have been automating other parts of your
```

File: src/content/basics/app-catalog/index.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+### What are Managed Apps?
+
+Managed Apps are provided by Giant Swarm and have a SLA associated with
+them. When you install one of our Managed Apps, we make sure that it stays
+running and even get paged if something goes wrong.
+
+Managed Apps can be found in the Managed _App Catalog_, indicated by the
+orange "Managed" badge.
+
+MORE DETAILS ON THE SLA AND EXPECTATION MANAGEMENT
```
Comment (puja108) [inline src/content/basics/app-catalog/index.md]: let's have a section on the different types of catalogs, i.e. managed, incubator, helm,... this whole section would be a sub-section of that

File: src/content/basics/app-catalog/index.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+### How can I interact with the App Catalog?
+
+Currently only through our web interface. See [this reference page] for more
+details on how to install and configure apps using our web interface.
+
+Support for the App Catalog is coming to our CLI (gsctl) next.
+
+Lastly, at the end of the day, the App Catalog is configured through a set of
+Custom Resources installed on your Control Plane.
```
Comment (puja108) [inline src/content/basics/app-catalog/index.md]: I know what you mean above and that this is linked to the para below, but it is not really clear. Maybe:

"Lastly, at the end of the day, what our interfaces do, is create (or update) a set of Custom Resources on your Control Plane."

File: src/content/basics/app-catalog/index.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+Our intention is to eventually grant you full access to the control plane
```
Comment (puja108) [inline src/content/basics/app-catalog/index.md]: here and in other places: you sometimes write "control plane" in lower case and sometimes use the capitalized version, we should stick with one (I tend to capitalize it)

File: src/content/basics/app-catalog/index.md (hunk 4 above)
Comment (oponder) [reply src/content/basics/app-catalog/index.md]: I think this kind of confuses things. Maybe when we add a section on the pre-installed apps that would be ok.

Comment (puja108) [reply src/content/basics/app-catalog/index.md]: Ok

File: src/content/basics/app-catalog/index.md (hunk 3 above)
Comment (oponder) [reply src/content/basics/app-catalog/index.md]: I did something wrong in the merging and brought this back... however, I don't think `an` is right grammar here.. ping @rossf7 for backup

> Consonant sounds call for an “a,” and vowel sounds call for an “an.”

Comment (rossf7) [reply src/content/basics/app-catalog/index.md]: Yes, can confirm it depends how the following word is spoken.

File: src/content/reference/web-interface/app-catalog.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+Once you know what app you'd like to install, click on that app, and then on
+"Configure & Install"
+
+![app-detail-page](../app-detail-page.png)
+
+That'll bring up a modal where you can choose what cluster you want to install
+the app on, as well as some further steps allowing you to configure the app.
+
+This is also where you can provide your values.yaml with custom configuration for
```
Comment (puja108) [inline src/content/reference/web-interface/app-catalog.md]: maybe link values.yaml to upstream helm values.yaml docs?

File: src/content/reference/web-interface/app-catalog.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+This page will give you an overview of how to do some common tasks related to the
+Giant Swarm App Catalog using our Web Interface.
+
+Go here for an [overview of the Giant Swarm App Catalog](/basics/app-catalog/) instead.
+
+### Viewing all App Catalogs
+
+Our web interface lets you browse the App Catalogs installed on your Control Plane.
+Click on "App Catalogs" in the navigation menu. The "App Catalogs" link will only
```
Comment (puja108) [inline src/content/reference/web-interface/app-catalog.md]: there's also the link in cluster details page

File: src/content/reference/web-interface/_index.md [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+## What can I do with the Web Interface?
+
+Our Web Interface let's you do the following things. Some topics link to more
+detailed information:
+
+- View and manage clusters
+- View and manage keypairs for your clusters
+- View and manage organizations
+- Manage your own account and password
+- Learn how to get started with your kubernetes clusters by following a guide
```
Comment (puja108) [inline src/content/reference/web-interface/_index.md]:
```suggestion
- Learn how to get started with your Kubernetes clusters by following a guide
```

File: src/content/reference/web-interface/app-catalog.md (hunk 11 above)
Comment (oponder) [reply src/content/reference/web-interface/app-catalog.md]: this? https://github.com/helm/helm/blob/master/docs/chart_template_guide/values_files.md

File: src/content/reference/web-interface/app-catalog.md (hunk 12 above)
Comment (oponder) [reply src/content/reference/web-interface/app-catalog.md]: True but, I think there's no need to mention that here as well. Users will find it.

File: src/content/basics/app-catalog/index.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+So to clarify a bit: the _Giant Swarm App Catalog_ refers to the whole feature,
+and an _App Catalog_ is a collection of _Apps_.
+
+We'll provide some _App Catalogs_, but so will our partners, and you too can set
+up your own internal catalog(s) for your whole company to use.
+
+### What makes up the Giant Swarm App Catalog?
+
+Technically the Giant Swarm App Catalog is implemented as a set of operators
+running on your Control Plane and tenant clusters. These operators watch various
```
Comment (rossf7) [inline src/content/basics/app-catalog/index.md]: I'm confused here. This sounds like the control plane catalog which I see as an internal catalog for us.

I think our Giant Swarm App Catalog has the core components we already run in tenant clusters. As well as future optional apps when they graduated. WDYT?

File: src/content/basics/app-catalog/index.md [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+Custom Resources, some created by us, and others created by you, which make up
+the desired state of the Giant Swarm App Catalog.
+
+For example, this "App" Custom Resource indicates that you want Instana installed
```
Comment (rossf7) [inline src/content/basics/app-catalog/index.md]: Minor but is there a reason we are using Instana here? I know they are friends but a more common example from community woud be better IMO.

File: src/content/basics/app-catalog/index.md [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+
+By default you will have the Incubator and the Community App Catalogs installed
+on your Control Planes.
+
+### The Giant Swarm Incubator App Catalog
+
+The Giant Swarm Incubator App Catalog contains apps that we would like to provide as
+a Managed App at some point in the future. When you install an app from the
+Giant Swarm Incubator App Catalog you are being brave and helping us get things
+right before we offer the SLA on that app.
```
Comment (rossf7) [inline src/content/basics/app-catalog/index.md]:
```suggestion
right before we offer an SLA on that app.
```

File: src/content/basics/app-catalog/index.md [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+The App Catalog CR is an abstraction, it's purpose is to point to _something_
+that contains a list of Apps and the storage location of where to get those Apps.
+
+Currently the only supported format for App Catalogs are
+[Helm Chart Repositories](https://github.com/helm/helm/blob/master/docs/chart_repository.md),
+however in the future we might support other ways of offering your Apps to the
+Giant Swarm App Catalog.
+
+We've created some automation and best practices around hosting a
+Helm Chart Repository on Github, see github.com/giantswarm/sample-catalog
```
Comment (rossf7) [inline src/content/basics/app-catalog/index.md]: I think sample-catalog is a bit rough in places. Should we link straight to the docs? https://github.com/giantswarm/sample-catalog#setting-up-a-new-app-catalog

File: src/content/basics/app-catalog/index.md (hunk 15 above)
Comment (oponder) [reply src/content/basics/app-catalog/index.md]: I'm happy to pick something else. Any suggestion?

Comment (rossf7) [reply src/content/basics/app-catalog/index.md]: What about grafana? Something that is a bit more generic.

**Pull request merged**